### ISSUE TITLE
Merlin transcript RNG

### ISSUE THREAD

**Issue by noot** [ChainSafe/go-schnorrkel#2]
Right now, sign and verify won't interoperate with the rust library, since in rust schnorrkel uses the transcript RNG to generate randomness when signing. The Go merlin library we're using doesn't have that yet: see https://github.com/gtank/merlin/issues/2

See:
https://github.com/w3f/schnorrkel/blob/798ab3e0813aa478b520c5cf6dc6e02fd4e07f0a/src/sign.rs#L161
https://github.com/w3f/schnorrkel/blob/798ab3e0813aa478b520c5cf6dc6e02fd4e07f0a/src/context.rs#L153 
for the relevant code in schnorrkel.

**Comment by oleganza** [comment]
Note that transcript-assisted RNG is not required for interoperability. Verifiers will never know how the signers generate nonces.

**Comment by noot** [comment]
@oleganza that is true, I should specify that it would be needed for a complete implementation.

**Comment by swdee** [comment]
I have run into a problem with Signing and Verifying output not working with the Rust library, specifically using Substrates `subkey` program to verify with and produce test vectors.

A simple test is here https://github.com/swdee/sr25519-test to explain the issue.

@noot  I see in the [gossamer code](https://github.com/ChainSafe/gossamer/blob/09c27233162157ff7d45f78fc1e7e4e28ec8050c/lib/crypto/sr25519/sr25519.go#L190)  its not doing anything different to my test script, and also the [test here](https://github.com/ChainSafe/go-schnorrkel/blob/d3c6d3118d10f4d5b906935f24e3b033190e66b6/sign_test.go#L92), so I would have expected this to work.

Is there something obvious I am missing here?
